src/pgduckdb_types.cpp

```diff
@@ -917,10 +918,41 @@ ConvertPostgresToBaseDuckColumnType(Form_pg_attribute &attribute) {
 
 duckdb::LogicalType
 ConvertPostgresToDuckColumnType(Form_pg_attribute &attribute) {
+	int dimensions = -1;
+	Oid save_typoid = InvalidOid;
+	std::lock_guard<std::mutex> lock(GlobalProcessLock::GetLock());
+
+	if (get_typtype(attribute->atttypid) == TYPTYPE_DOMAIN) {
+		/* If the domain is an array type, you need to obtain the corresponding array dimension information */
+		if (type_is_array_domain(attribute->atttypid)) {
+			HeapTuple typeTuple = SearchSysCache1(TYPEOID, ObjectIdGetDatum(attribute->atttypid));
+			dimensions = ((Form_pg_type)GETSTRUCT(typeTuple))->typndims;
+			ReleaseSysCache(typeTuple);
+		}
+
+		save_typoid = attribute->atttypid;
+		/* It is a domain type that needs to be reduced to its base type */
+		attribute->atttypid = getBaseType(attribute->atttypid);
```

Comment:
Instead of changing the the `attribute` argument in place, let's change `ConvertPostgresToBaseDuckColumnType` to take an `atttypid` and an `attypemod` argument. Then we can mark the `attribute` argument as const.